For 517 2

merschel opened the pull request.

merschel committed:
FOR-517 Adds unit tests for compression and decompression

Introduces unit tests for compression and decompression functionalities.

This includes tests for:
- Timestamp compression and decompression with various scenarios
- Sensor value compression and decompression, covering different data patterns
- Data validation during the compression process.

These tests aim to ensure the robustness and correctness of the compression and decompression algorithms.

merschel committed:
FOR-517 Adds data validation rules and tests

Adds data validation rules for compression and decompression to the README.

The validation ensures data integrity by checking for the presence of Unix timestamps, matching sensor data lengths,
and proper section sizes and durations during compression and decompression.

Introduces new unit tests for validating data during compression and decompression processes.
These tests cover scenarios like missing timestamps, mismatched sensor data lengths,
and invalid section sizes or durations, ensuring data integrity.

merschel committed:
Merge branch 'main' into FOR-517_2

piperdiana commented on unittest/decompression/decompressionTimestamps:
```diff
@@ -0,0 +1,122 @@
+
+### 1. ZeroTimestampTest
+
+- **Purpose**: Tests timestamps comression with empty unix timestamps array
```
decompression statt compression

merschel replied:
done

piperdiana commented on unittest/compression/compressionValues.md:
```diff
@@ -0,0 +1,162 @@
+
+### 20. PPGValuesTest
+
+- **Purpose**: Tests delta encoding with values similar to a ppg signal
```
wäre hier nicht Sinnvoller Werte von einem echten PPG Beispiel, eventuell auch ein mit großen Sprünge?

merschel replied:
done

merschel committed:
FOR-517 Fixes typos in timestamp compression tests

Corrects minor typographical errors in the descriptions of the timestamp compression and decompression tests.

This improves the readability and clarity of the test documentation.

merschel committed:
FOR-517 Updates compression validation and tests

Adds validation for time zone offset during compression and
decompression.

Updates compression unit tests to include real-world sensor data
for PPG, ACC, and ECG signals, improving test coverage and
realism.
